MetaMask/metamask-extension
# fix: OpenSea supply -> total_supply

## **Description**

OpenSea renamed the field `supply`, which used to be required, to `total_supply`.  Fixes the error:

`undefined is not an object (evaluating 'contract.supply.toString')`

A more accurate `total_supply` was also added to the collection object, which is now preferred to the one on the contract.

## **Related issues**


## **Manual testing steps**

1. Import a wallet containing NFTs on mainnet
2. Enable NFT autodetection in settings, verify the NFTs appear
3. Verify descriptions, collection/token images are correct

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
